### PR TITLE
Mustang/Collections

### PR DESCRIPTION
**Pull request by willdoran**
This pull request makes the following changes:
- Adding collection post toggle directive
- Adding directive for add to collection button
- Adding directive for add to collection link

Test these changes by:
-

Fixes ushahidi/platform# .

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform-client/200)

### CODE REVIEW

**Comment by rjmackay** [comment]
Looking pretty good.
Needs a few changes to naming just to aid clarity when reading the code.

Theres some pretty ugly/complex code in some parts but I think thats mostly the legacy of how we did this before - making a mental note to revisit but not right now.

<details><summary><i>Previously, coveralls wrote…</i></summary>
> 
> [Coverage Status](https://coveralls.io/builds/6351638)
> 
> Coverage decreased (-1.5%) to 55.871% when pulling **e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa on mustang/add-to-collection** into **72683c853943bb42983aa5dd98de3613ae188a4d on develop**.
> 
</details>


---

Reviewed 3 of 15 files at r1, 3 of 5 files at r3, 2 of 2 files at r6, 20 of 21 files at r7, 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 14 unresolved discussions, some commit checks failed.

---

*[app/common/directives/modal.js, line 34 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIk6eKZcUIqpHU9UeV5:-KIk6eKZcUIqpHU9UeV6:-455555920) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/common/directives/modal.js#L34)):*
> ```JavaScript
>             $scope.classVisible = false;
>             $scope.modalOffset = 0;
>             $scope.title = $translate($scope.title);
> ```

I don't think this works. $translate returns a promise. You can either use $translate.instant() or handle the promise. Not 100% sure how instant() works though

---

*[app/common/directives/collection-toggle/collection-toggle-button.js, line 20 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIk79J75v6EtWM64bfE:-KIk79J75v6EtWM64bfF:-1382959498) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/common/directives/collection-toggle/collection-toggle-button.js#L20)):*
> ```JavaScript
>             $scope.toggleCollection = function () {
> 
>                 var postsSet = _.filter($scope.posts, function (post) {
> ```

I'm not clear: what does postsSet contains?  full posts objects?
Can you give this a clear name ie. selectedPosts

---

*[app/common/directives/collection-toggle/collection-toggle-button.js, line 24 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIk7Qckx4vB8WY6WZjA:-KIk7Qckx4vB8WY6WZjB:-1338682331) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/common/directives/collection-toggle/collection-toggle-button.js#L24)):*
> ```JavaScript
>                 });
> 
>                 $rootScope.$emit('event:collection:show:toggle', postsSet);
> ```

is it showing or toggling? event name is confusing.

---

*[app/common/directives/collection-toggle/collection-toggle-link.js, line 3 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIk7hCx1y0K9ku0Pw0d:-KIk7hCyNwNxFfBsjHod:-1631325428) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/common/directives/collection-toggle/collection-toggle-link.js#L3)):*
> ```JavaScript
> module.exports = [
>     '$rootScope',
>     '$translate',
> ```

unneeded?

---

*[app/set/controllers/collections-controller.js, line 35 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIk8ICAUqdZP8n0irHt:-KIk8ICBcu6yxb7DdBs4:-1265652795) ([raw file](https://github.com/ushahidi/platform-client/blob/72683c853943bb42983aa5dd98de3613ae188a4d/app/set/controllers/collections-controller.js#L35)):*
> ```JavaScript
> 
>         // Check if we can edit
>         $scope.canEdit = function () {
> ```

Where's this logic moved to?

---

*[app/set/directives/collections/listing.js, line 38 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkJkqevbg-JIxq8lmS:-KIkJkqevbg-JIxq8lmT:-327939530) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/listing.js#L38)):*
> ```JavaScript
> 
>             $scope.toggleCollection = function (selectedCollection) {
>                 if ($scope.posts.length === 1) {
> ```

I now understand what you were talking about on our last call. Having to switch on the length of the posts argument is kinda gross. Not much we can do about it. Though some comments would really help.

---

*[app/set/directives/collections/listing.js, line 134 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkAoJcm0EQ0U--4FbI:-KIkAoJcm0EQ0U--4FbJ:1578124486) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/listing.js#L134)):*
> ```JavaScript
>             });
> 
>             $rootScope.$on('event:collection:show:toggle:after:create', function (event, posts, collection) {
> ```

Can we shorten this event name please??
1. do they really need an `event:` prefix? we know they're events
2. limit them to 2-3 levels. ie. `collection:show` `collection:afterCreate` .. `:` should be a namespace, not simply a separator between every word

---

*[app/set/directives/collections/mode-context.js, line 1 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkGrew1kJd1b1FqkhY:-KIkGrew1kJd1b1FqkhZ:-1640968862) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/mode-context.js#L1)):*
> ```JavaScript
> module.exports = CollectionModeContext;
> ```

I'm a bit torn - but I *think* the file name should match this: ie collection-mode-context.directive.js

---

*[app/set/directives/collections/mode-context.js, line 16 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkGWiT6RvT_Vlk-o7A:-KIkGWiT6RvT_Vlk-o7B:1837742986) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/mode-context.js#L16)):*
> ```JavaScript
>  }
> 
> ModeContextController.$inject = [
> ```

Can you name this `CollectionModeContextController` ?
part of why this style is useful is that the function name will now show up in the debugger/exceptions. So its useful to know we're seeing an error from the CollectionModeContext, not just any ModeContext 

---

*[app/set/directives/collections/mode-context.js, line 44 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkH-4dCV1BkrXkV0c1:-KIkH-4dCV1BkrXkV0c2:380399470) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/mode-context.js#L44)):*
> ```JavaScript
> 
>     function activate() {
>         $scope.saveNotification = saveNotification;
> ```

why are these inside activate?

---

*[app/set/directives/collections/mode-context.js, line 47 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkH46zc1RgDxl3cK8L:-KIkH46zc1RgDxl3cK8M:-48486462) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/mode-context.js#L47)):*
> ```JavaScript
>         $scope.removeNotification = removeNotification;
>         // Show Add Notification link
>         $scope.showNotificationLink = false;
> ```

initial state for this should be set outside of activate, even if you do set it again in activate

---

*[app/set/directives/collections/mode-context.js, line 56 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkHG5Oai_B6wACKr_7:-KIkHG5Oai_B6wACKr_8:1185898267) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/mode-context.js#L56)):*
> ```JavaScript
>         });
> 
>         $scope.canEditCollection = canEdit($scope.collection);
> ```

Could this move to the top of the function? or at least declare a default value.

---

*[app/set/directives/collections/mode-context.js, line 59 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkHo0XUYHIKBHGYaWe:-KIkHo0XUYHIKBHGYaWf:-861171285) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/mode-context.js#L59)):*
> ```JavaScript
>     }
> 
>     function canEdit (collection) {
> ```

pretty sure you'll get jscs/jshint errors on this. it should be
`function canEdit(collection) {`
note: no space after function name

---

*[server/www/templates/sets/collections/collection-listing.html, line 16 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkIQ86JxNo7lHKqBL7:-KIkIQ86JxNo7lHKqBL8:-1710060993) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/server/www/templates/sets/collections/collection-listing.html#L16)):*
> ```HTML
> 
>                 <div class="listing-item" ng-repeat="collection in collections">
>                     <div class="listing-item-select" ng-if="isToggleMode">
> ```

I assume ng-if isn't caused side effects here due to the child scope?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-:-KIkKN4LlVcqHSgx06ZM:-1375766843)*

**Comment by willdoran** [comment]
Review status: all files reviewed at latest revision, 14 unresolved discussions, some commit checks failed.

---

*[app/common/directives/modal.js, line 34 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIk6eKZcUIqpHU9UeV5:-KIn3QGYs0fEZ8CH-jaI:-306505350) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/common/directives/modal.js#L34)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> I don't think this works. $translate returns a promise. You can either use $translate.instant() or handle the promise. Not 100% sure how instant() works though
</details>
It should return a promise but the idea will be set when that promise is completed by the digest cycle. BUT I've changed it to be instant which is synchronous.

---

*[app/common/directives/collection-toggle/collection-toggle-button.js, line 20 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIk79J75v6EtWM64bfE:-KIn3b1dXKXMRrS4sDWM:210451299) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/common/directives/collection-toggle/collection-toggle-button.js#L20)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> I'm not clear: what does postsSet contains?  full posts objects?
> Can you give this a clear name ie. selectedPosts
</details>
Yep, it is the set of post objects that have been selected, I'll call is selectedPostObjects

---

*[app/common/directives/collection-toggle/collection-toggle-button.js, line 24 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIk7Qckx4vB8WY6WZjA:-KIn3ug_XJ0lG9Y7N2JL:1982720968) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/common/directives/collection-toggle/collection-toggle-button.js#L24)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> is it showing or toggling? event name is confusing.
</details>
That is legacy because of there was previously a toggle directive which is now integrated with listing. I'll reorder it according to the event string format you suggested below.

---

*[app/common/directives/collection-toggle/collection-toggle-link.js, line 3 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIk7hCx1y0K9ku0Pw0d:-KIn4OxsmK7QH9RHB37-:-1064776208) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/common/directives/collection-toggle/collection-toggle-link.js#L3)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> unneeded?
</details>
Nope it's used to set the link text, add_to_collection

---

*[app/set/controllers/collections-controller.js, line 35 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIk8ICAUqdZP8n0irHt:-KIn4VrXkWOrnL6zgTeN:1392704848) ([raw file](https://github.com/ushahidi/platform-client/blob/72683c853943bb42983aa5dd98de3613ae188a4d/app/set/controllers/collections-controller.js#L35)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Where's this logic moved to?
</details>
It has moved to the mode-context.js directive

---

*[app/set/directives/collections/listing.js, line 38 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkJkqevbg-JIxq8lmS:-KIn4_Uk0S_a7Ixnl4P5:1446780294) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/listing.js#L38)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> I now understand what you were talking about on our last call. Having to switch on the length of the posts argument is kinda gross. Not much we can do about it. Though some comments would really help.
</details>
I'll comment out the flow, may be it can be improved in future, it's a very common use case to want to pass either many or one there should be a pattern we can find that will reduce the grossness.

---

*[app/set/directives/collections/listing.js, line 134 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkAoJcm0EQ0U--4FbI:-KIn6zJHYamjlsxtlIk6:892390124) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/listing.js#L134)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Can we shorten this event name please??
> 1. do they really need an `event:` prefix? we know they're events
> 2. limit them to 2-3 levels. ie. `collection:show` `collection:afterCreate` .. `:` should be a namespace, not simply a separator between every word
</details>
Yeah - if we had 5 types then yeah event would make sense but eventually we should have a data layer so this should go away.
ok, I'll do noun:action:<optional_action>

---

*[app/set/directives/collections/mode-context.js, line 1 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkGrew1kJd1b1FqkhY:-KIn9T1DqCgbSs--wNOZ:-2022477658) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/mode-context.js#L1)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> I'm a bit torn - but I *think* the file name should match this: ie collection-mode-context.directive.js
</details>
Ok, so I'm happy to go whatever route that is consistent BUT I want to make an argument for why I made this choice.

I agree with making it mode-context.directive.js BUT
I think how you would read this is
I care about the mode context within collections within sets
sets/collections/mode-context.directive.js

Adding an extra collections in front of mode-context - to me - would imply it should just be in the sets directory BUT I think that would make the sets directory look messier. I'm struggling to decide how subjective this is on my part but I feel strongly that this a cleaner looking structure.

---

*[app/set/directives/collections/mode-context.js, line 16 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkGWiT6RvT_Vlk-o7A:-KInCU6eNMN751fwV9mj:-499146009) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/mode-context.js#L16)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Can you name this `CollectionModeContextController` ?
> part of why this style is useful is that the function name will now show up in the debugger/exceptions. So its useful to know we're seeing an error from the CollectionModeContext, not just any ModeContext 
</details>
Done.

---

*[app/set/directives/collections/mode-context.js, line 44 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkH-4dCV1BkrXkV0c1:-KInCWvma-03b5gf7oNI:-1197793996) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/mode-context.js#L44)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> why are these inside activate?
</details>
Done.

Where should $rootScope.$on/$emit go - separate section?

---

*[app/set/directives/collections/mode-context.js, line 47 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkH46zc1RgDxl3cK8L:-KInCY-kaecdfOswx4XW:-499146009) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/mode-context.js#L47)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> initial state for this should be set outside of activate, even if you do set it again in activate
</details>
Done.

---

*[app/set/directives/collections/mode-context.js, line 56 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkHG5Oai_B6wACKr_7:-KInCYpoXx7sak39OJ-i:-499146009) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/mode-context.js#L56)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Could this move to the top of the function? or at least declare a default value.
</details>
Done.

---

*[app/set/directives/collections/mode-context.js, line 59 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkHo0XUYHIKBHGYaWe:-KInCaEiCupxONbja8H_:-1582175016) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/mode-context.js#L59)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> pretty sure you'll get jscs/jshint errors on this. it should be
> `function canEdit(collection) {`
> note: no space after function name
</details>
It passes linting for me but I'll change it for consistency

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/200)*

File: app/common/directives/modal.js [hunk 1]
```diff
@@ -26,9 +28,10 @@ function (
             $scope.iconPath = $scope.icon ? '../../img/iconic-sprite.svg#' + $scope.icon : '';
 
             var classChangePromise = null;
-
+            $scope.iconPath = '../../img/iconic-sprite.svg#' + $scope.icon;
             $scope.classVisible = false;
             $scope.modalOffset = 0;
+            $scope.title = $translate.instant($scope.title);
```
Comment (rjmackay) [inline app/common/directives/modal.js]: Hrm..  If it worked with the promise thats fine, but the docs for ngtranslate didn't seem like it work. ngResource does some special magic with its promises that makes this sort thing work on endpoints, but I thought that was a special case.

File: app/common/directives/collection-toggle/collection-toggle-link.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+module.exports = [
+    '$rootScope',
+    '$translate',
+function (
+    $rootScope,
+    $translate
```
Comment (rjmackay) [inline app/common/directives/collection-toggle/collection-toggle-link.js]: Not needed. There are no $translate calls in this file at all. You don't need to inject the service to use the translate directives in your template.

File: app/set/directives/collections/listing.directive.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    // Show listing modal
+    $rootScope.$on('collectionListing:show', function () {
+        $scope.collectionListingVisible = true;
+        $scope.loadCollections();
+    });
+
+    // Show listing modal in Collection Toggle mode
+    // This allows users to select collection to add a given post(s) to
+    // This is called with a post array of 1 or more entries
+    $rootScope.$on('collectionToggle:show', function (event, posts) {
```
Comment (rjmackay) [inline app/set/directives/collections/listing.directive.js]: For future : I've generally be putting event bindings before all other functions. Often as named functions so you can scan the file quickly and find
```
$rootScope.$on('collectionToggle:show', openCollectionToggle);
```

**Comment by rjmackay** [comment]
Good to land - I had a few more minor comments but they're lost to reviewable's stupidity.

**Comment by rjmackay** [comment]
LGTM

<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Looking pretty good.
> Needs a few changes to naming just to aid clarity when reading the code.
> 
> Theres some pretty ugly/complex code in some parts but I think thats mostly the legacy of how we did this before - making a mental note to revisit but not right now.
</details>


---

Reviewed 2 of 23 files at r7, 11 of 13 files at r9.
Review status: 26 of 27 files reviewed at latest revision, 13 unresolved discussions, some commit checks failed.

---

*[app/common/directives/modal.js, line 34 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIk6eKZcUIqpHU9UeV5:-KIo1twrmEh8V_Dfjux6:-1098067085) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/common/directives/modal.js#L34)):*
<details><summary><i>Previously, willdoran (will) wrote…</i></summary>
> It should return a promise but the idea will be set when that promise is completed by the digest cycle. BUT I've changed it to be instant which is synchronous.
</details>
Hrm..  If it worked with the promise thats fine, but the docs for ngtranslate didn't seem like it work. ngResource does some special magic with its promises that makes this sort thing work on endpoints, but I thought that was a special case.

---

*[app/common/directives/collection-toggle/collection-toggle-link.js, line 3 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIk7hCx1y0K9ku0Pw0d:-KIo2R-w_6IDeRerBEdM:-1695540520) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/common/directives/collection-toggle/collection-toggle-link.js#L3)):*
<details><summary><i>Previously, willdoran (will) wrote…</i></summary>
> Nope it's used to set the link text, add_to_collection
</details>
huh? but  there are not $translate calls in this file at all? You don't need to inject the service to use the translate directives in your template.

---

*[app/set/directives/collections/editor.directive.js, line 43 \[r9\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIo2rc5OoddpzoMzzZ9:-KIo2rc5OoddpzoMzzZA:829436904) ([raw file](https://github.com/ushahidi/platform-client/blob/0929538f3b0d9a1e033f9404aac6a1b1e2fcf07d/app/set/directives/collections/editor.directive.js#L43)):*
> ```JavaScript
>     $scope.views = ViewHelper.views();
> 
>     $scope.setBasicCollection = setBasicCollection;
> ```

so much cleaner :+1: 

---

*[app/set/directives/collections/listing.directive.js, line 185 \[r9\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIo3CXwviB12i56INfr:-KIo3CXwviB12i56INfs:1160627969) ([raw file](https://github.com/ushahidi/platform-client/blob/0929538f3b0d9a1e033f9404aac6a1b1e2fcf07d/app/set/directives/collections/listing.directive.js#L185)):*
> ```JavaScript
>     // This allows users to select collection to add a given post(s) to
>     // This is called with a post array of 1 or more entries
>     $rootScope.$on('collectionToggle:show', function (event, posts) {
> ```

For future : I've generally be putting event bindings before all other functions. Often as named functions so you can scan the file quickly and find
```
$rootScope.$on('collectionToggle:show', openCollectionToggle);
```

---

*[app/set/directives/collections/mode-context.directive.js, line 1 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkGrew1kJd1b1FqkhY:-KIo487WI2YWhTAeaVgN:-1422660987) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/mode-context.directive.js#L1)):*
<details><summary><i>Previously, willdoran (will) wrote…</i></summary>
> Ok, so I'm happy to go whatever route that is consistent BUT I want to make an argument for why I made this choice.
> 
> I agree with making it mode-context.directive.js BUT
> I think how you would read this is
> I care about the mode context within collections within sets
> sets/collections/mode-context.directive.js
> 
> Adding an extra collections in front of mode-context - to me - would imply it should just be in the sets directory BUT I think that would make the sets directory look messier. I'm struggling to decide how subjective this is on my part but I feel strongly that this a cleaner looking structure.
> 
</details>
OK makes sense. We're really inconsistent about this so we should reconsider later and make a final choice.

---

*[app/set/directives/collections/mode-context.directive.js, line 44 \[r8\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIkH-4dCV1BkrXkV0c1:-KIo4VYXFqDCye_1X7hK:-298495575) ([raw file](https://github.com/ushahidi/platform-client/blob/e7d7015904d2fa42f41b04d5c42c6b6c7bf5d0fa/app/set/directives/collections/mode-context.directive.js#L44)):*
<details><summary><i>Previously, willdoran (will) wrote…</i></summary>
> Done.
> 
> Where should $rootScope.$on/$emit go - separate section?
</details>
I've been keeping the $on bindings near the top. Not perfect but good enough right now

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-:-KIo4yLUo1jmTo7XwCUK:676834233)*

**Comment by rjmackay** [comment]
eep. Just noticed the logic for actually pushing the collection ID onto the post filters is now AWOL. That said: I can fix that up while I wire up saved searches.

<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> LGTM
</details>


---

Review status: 26 of 27 files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---

*[app/set/controllers/collections-controller.js, line 49 \[r9\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIzrjikvxAjr2WNeWNo:-KIzrjiliMfRlMla37Wd:-1735549205) ([raw file](https://github.com/ushahidi/platform-client/blob/72683c853943bb42983aa5dd98de3613ae188a4d/app/set/controllers/collections-controller.js#L49)):*
> ```JavaScript
>         };
> 
>         // Extend filters, always adding the current collection id
> ```

I think this bit of logic has gone missing

---

*[app/set/controllers/collections-controller.js, line 58 \[r9\]](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-KIzrm9oLXrOAq6gq8uU:-KIzrm9oLXrOAq6gq8uV:-373693827) ([raw file](https://github.com/ushahidi/platform-client/blob/72683c853943bb42983aa5dd98de3613ae188a4d/app/set/controllers/collections-controller.js#L58)):*
> ```JavaScript
>         // whenever the GlobalFilter post query changes,
>         // update the current list of posts
>         $scope.$watch(function () {
> ```

.. and this is missing too?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/200#-:-KIzs1MQzBj910Mogxjd:-2056574558)*

**Pull request merged**